source/expressions.tex

```diff
@@ -1422,8 +1422,14 @@
 \end{note}
 
 \pnum
-A \defnadj{component}{name} of an \grammarterm{unqualified-id} $U$ is $U$
-if it is a name or the component name of the \grammarterm{template-id} or \grammarterm{type-name} of $U$, if any.
+A \defn{component name} of an \grammarterm{unqualified-id} $U$ is
+\begin{itemize}
+\item
+$U$ if it is a name or
+\item
+the component name of
+the \grammarterm{template-id} or \grammarterm{type-name} of $U$, if any.
```

Comment:
What if there aren't any?

Reply:
Then the set of component names is empty, I guess.